### PR TITLE
python3Packages.torch{audio, vision}: update

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Update the following packages to their latest version:
- `python3Packages.torch`: 2.0.0 -> 2.0.1
- `python3Packages.torchaudio`: 2.0.1 -> 2.0.2
- `python3Packages.torchvision`: 1.15.1 -> 1.15.2

cc @samuela @junjihashimoto @SomeoneSerge 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by junjihashimoto — APPROVED** [review]
LGTM!

**Review by junjihashimoto — APPROVED** [review]
LGTM!

**Comment by samuela** [comment]
Result of `nixpkgs-review pr 232412` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>71 packages failed to build:</summary>
  <ul>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>python310Packages.ax</li>
    <li>python310Packages.ax.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.botorch</li>
    <li>python310Packages.botorch.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.deepwave</li>
    <li>python310Packages.deepwave.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.flammkuchen</li>
    <li>python310Packages.flammkuchen.dist</li>
    <li>python310Packages.gpytorch</li>
    <li>python310Packages.gpytorch.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.linear_operator</li>
    <li>python310Packages.linear_operator.dist</li>
    <li>python310Packages.pymanopt</li>
    <li>python310Packages.pymanopt.dist</li>
    <li>python310Packages.pytorch-metric-learning</li>
    <li>python310Packages.pytorch-metric-learning.dist</li>
    <li>python310Packages.pywick</li>
    <li>python310Packages.pywick.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-machine-learning</li>
    <li>python310Packages.qiskit-machine-learning.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.sfepy</li>
    <li>python310Packages.sfepy.dist</li>
    <li>python310Packages.shap</li>
    <li>python310Packages.shap.dist</li>
    <li>python310Packages.skorch</li>
    <li>python310Packages.skorch.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.slicer</li>
    <li>python310Packages.slicer.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.timm</li>
    <li>python310Packages.timm.dist</li>
    <li>python310Packages.tiny-cuda-nn</li>
    <li>python310Packages.torchWithCuda</li>
    <li>python310Packages.torchWithCuda.dev</li>
    <li>python310Packages.torchWithCuda.dist</li>
    <li>python310Packages.torchWithCuda.lib</li>
    <li>python310Packages.torchWithRocm</li>
    <li>python310Packages.torchWithRocm.dev</li>
    <li>python310Packages.torchWithRocm.dist</li>
    <li>python310Packages.torchWithRocm.lib</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
  </ul>
</details>
<details>
  <summary>112 packages built:</summary>
  <ul>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>larynx-train (python310Packages.larynx-train)</li>
    <li>larynx-train.dist (python310Packages.larynx-train.dist)</li>
    <li>openai-whisper (python310Packages.openai-whisper)</li>
    <li>openai-whisper.dist (python310Packages.openai-whisper.dist)</li>
    <li>pianotrans</li>
    <li>pianotrans.dist</li>
    <li>python310Packages.accelerate</li>
    <li>python310Packages.accelerate.dist</li>
    <li>python310Packages.apptools</li>
    <li>python310Packages.apptools.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bitsandbytes</li>
    <li>python310Packages.bitsandbytes.dist</li>
    <li>python310Packages.blosc2</li>
    <li>python310Packages.blosc2.dist</li>
    <li>python310Packages.boxx</li>
    <li>python310Packages.boxx.dist</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.clip</li>
    <li>python310Packages.clip.dist</li>
    <li>python310Packages.deepdish</li>
    <li>python310Packages.deepdish.dist</li>
    <li>python310Packages.envisage</li>
    <li>python310Packages.envisage.dist</li>
    <li>python310Packages.ezyrb</li>
    <li>python310Packages.ezyrb.dist</li>
    <li>python310Packages.ffcv</li>
    <li>python310Packages.ffcv.dist</li>
    <li>python310Packages.grad-cam</li>
    <li>python310Packages.grad-cam.dist</li>
    <li>python310Packages.ignite</li>
    <li>python310Packages.ignite.dist</li>
    <li>python310Packages.invisible-watermark</li>
    <li>python310Packages.invisible-watermark.dist</li>
    <li>python310Packages.kornia</li>
    <li>python310Packages.kornia.dist</li>
    <li>python310Packages.lion-pytorch</li>
    <li>python310Packages.lion-pytorch.dist</li>
    <li>python310Packages.manifest-ml</li>
    <li>python310Packages.manifest-ml.dist</li>
    <li>python310Packages.mayavi</li>
    <li>python310Packages.mayavi.dist</li>
    <li>python310Packages.mmcv</li>
    <li>python310Packages.mmcv.dist</li>
    <li>python310Packages.monai</li>
    <li>python310Packages.monai.dist</li>
    <li>python310Packages.nianet</li>
    <li>python310Packages.nianet.dist</li>
    <li>python310Packages.pandas-stubs</li>
    <li>python310Packages.pandas-stubs.dist</li>
    <li>python310Packages.peft</li>
    <li>python310Packages.peft.dist</li>
    <li>python310Packages.piano-transcription-inference</li>
    <li>python310Packages.piano-transcription-inference.dist</li>
    <li>python310Packages.pydmd</li>
    <li>python310Packages.pydmd.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pyro-ppl</li>
    <li>python310Packages.pyro-ppl.dist</li>
    <li>python310Packages.pytorch-lightning</li>
    <li>python310Packages.pytorch-lightning.dist</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.pytorch-pfn-extras.dist</li>
    <li>python310Packages.rising</li>
    <li>python310Packages.rising.dist</li>
    <li>python310Packages.safetensors</li>
    <li>python310Packages.safetensors.dist</li>
    <li>python310Packages.sentence-transformers</li>
    <li>python310Packages.sentence-transformers.dist</li>
    <li>python310Packages.stanza</li>
    <li>python310Packages.stanza.dist</li>
    <li>python310Packages.tables</li>
    <li>python310Packages.tables.dist</li>
    <li>python310Packages.tensorboardx</li>
    <li>python310Packages.tensorboardx.dist</li>
    <li>python310Packages.test-tube</li>
    <li>python310Packages.test-tube.dist</li>
    <li>python310Packages.torch (python310Packages.torchWithoutCuda ,python310Packages.torchWithoutRocm)</li>
    <li>python310Packages.torch-tb-profiler</li>
    <li>python310Packages.torch-tb-profiler.dist</li>
    <li>python310Packages.torch.dev (python310Packages.torchWithoutCuda.dev ,python310Packages.torchWithoutRocm.dev)</li>
    <li>python310Packages.torch.dist (python310Packages.torchWithoutCuda.dist ,python310Packages.torchWithoutRocm.dist)</li>
    <li>python310Packages.torch.lib (python310Packages.torchWithoutCuda.lib ,python310Packages.torchWithoutRocm.lib)</li>
    <li>python310Packages.torchaudio</li>
    <li>python310Packages.torchaudio.dist</li>
    <li>python310Packages.torchgpipe</li>
    <li>python310Packages.torchgpipe.dist</li>
    <li>python310Packages.torchinfo</li>
    <li>python310Packages.torchinfo.dist</li>
    <li>python310Packages.torchio</li>
    <li>python310Packages.torchio.dist</li>
    <li>python310Packages.torchlibrosa</li>
    <li>python310Packages.torchlibrosa.dist</li>
    <li>python310Packages.torchmetrics</li>
    <li>python310Packages.torchmetrics.dist</li>
    <li>python310Packages.torchvision</li>
    <li>python310Packages.torchvision.dist</li>
    <li>python310Packages.trainer</li>
    <li>python310Packages.trainer.dist</li>
    <li>python310Packages.ttach</li>
    <li>python310Packages.ttach.dist</li>
    <li>python310Packages.zcs</li>
    <li>python310Packages.zcs.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>

**Comment by samuela** [comment]
###### Failed derivations

- [ ] python3.10-hickle-5.0.2.drv (derivation hash: 73v9qz4v9mzydp69wx3v26g9f4yzz4nx)
  - https://gist.github.com/samuela/1251de100a836ade3fc6097dcc3bcb23
- [ ] python3.10-flax-0.6.5.drv (derivation hash: 70dnsz5j38zlqygmfshr6g630jp3r1pk)
  - https://gist.github.com/samuela/864caee9b24a052c7e8ea5f2c4835e1c
- [ ] python3.10-pims-0.6.1.drv (derivation hash: 6ky1hr0bb90wiwkrzxdvjcbc0vmrp981)
  - https://gist.github.com/samuela/631799ad3ddc49d09534ed71202ec3a7
- [ ] python3.10-qiskit-terra-0.21.0.drv (derivation hash: i3chvn8lpdaxn6wj88z9y75xchry7d39)
  - https://gist.github.com/samuela/5a326597b40a7cd70b0537ff1739fdbe
- [ ] python3.10-pymanopt-2.1.1.drv (derivation hash: rvdzqinkb451v8cp1cpwiwz6zanacqa2)
  - https://gist.github.com/samuela/42391cd098fa5355897af3dd1b6b3fd2
- [ ] tiny-cuda-nn.drv (derivation hash: x1hy0mahw14mcbjwhliq0bl6p35gp9cc)
  - https://gist.github.com/samuela/378f3aa2f7514797420c06425191f90d
- [ ] python3.10-slicer-0.0.7.drv (derivation hash: xbhq05nxmm9f4ddkjy1rfcmdkidmh853)
  - https://gist.github.com/samuela/3e54d8e82373e1bc04703e7405f29fac
- [ ] python3.10-skorch-0.12.1.drv (derivation hash: qll128r0djswdd91b90hzldy8xgsvkyc)
  - https://gist.github.com/samuela/8cc21bb5abf164defd8c575482940b4f
- [ ] python3.10-wandb-0.15.2.drv (derivation hash: bljn64bw02naljri98c3j59vac0yhxgk)
  - https://gist.github.com/samuela/66dea36016457ba3b25853b46783fc61
- [ ] python3.10-flammkuchen-1.0.2.drv (derivation hash: x590yl8i7d3qnm9kjpxq6j8d98gbqgcy)
  - https://gist.github.com/samuela/e25b2de48cf1d4a76bd72c983aec9bb5

**Comment by samuela** [comment]
###### Failed derivations

- [ ] python3.10-hickle-5.0.2.drv (derivation hash: 73v9qz4v9mzydp69wx3v26g9f4yzz4nx)
  - https://gist.github.com/samuela/58733f5ceafe441ff0348270c635850b
- [ ] python3.10-flax-0.6.5.drv (derivation hash: 70dnsz5j38zlqygmfshr6g630jp3r1pk)
  - https://gist.github.com/samuela/cd00a4019cb016bfebf9a98408ff32c5
- [ ] python3.10-pims-0.6.1.drv (derivation hash: 6ky1hr0bb90wiwkrzxdvjcbc0vmrp981)
  - https://gist.github.com/samuela/cee1b609921b168d6e8e36ae7a9c8e7a
- [ ] python3.10-qiskit-terra-0.21.0.drv (derivation hash: i3chvn8lpdaxn6wj88z9y75xchry7d39)
  - https://gist.github.com/samuela/79945f5de32daba0a31f8f31d10e8e8c
- [ ] python3.10-pymanopt-2.1.1.drv (derivation hash: rvdzqinkb451v8cp1cpwiwz6zanacqa2)
  - https://gist.github.com/samuela/815551670104b5570247821f2a69f526
- [ ] tiny-cuda-nn.drv (derivation hash: x1hy0mahw14mcbjwhliq0bl6p35gp9cc)
  - https://gist.github.com/samuela/190f0d1832ec2717f694ee91135d5d62
- [ ] python3.10-slicer-0.0.7.drv (derivation hash: xbhq05nxmm9f4ddkjy1rfcmdkidmh853)
  - https://gist.github.com/samuela/b45e94ecd9d5df6c3b69305600c6d925
- [ ] python3.10-skorch-0.12.1.drv (derivation hash: qll128r0djswdd91b90hzldy8xgsvkyc)
  - https://gist.github.com/samuela/64ee3275f1b9dfcec6e94c628f7d59ea
- [ ] python3.10-wandb-0.15.2.drv (derivation hash: bljn64bw02naljri98c3j59vac0yhxgk)
  - https://gist.github.com/samuela/084ea45985de77763d3b1ae6d2b8c7de
- [ ] python3.10-flammkuchen-1.0.2.drv (derivation hash: x590yl8i7d3qnm9kjpxq6j8d98gbqgcy)
  - https://gist.github.com/samuela/78bd32b6aec1810ca92fb794c9b152db

**Comment by samuela** [comment]
not sure why wandb failed in nixpkgs-review. i checked out this branch and it built fine

**Review by samuela — APPROVED** [review]

**Comment by samuela** [comment]
thanks @GaetanLepage !
